### PR TITLE
Cache dirs

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Align cache directories.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/165)

### CODE REVIEW

**Comment by sschuberth** [comment]
These changes are good, but shouldn't we also change

    OkHttpClientHelper.execute("scanner", request)

to

    OkHttpClientHelper.execute("scanner/http", request)

then, even if there's not yet another cache than "http" for the scanner?

**Comment by sschuberth** [comment]
Thinking more about this, I guess I'd prefer a structure like

    ~/.ort/analyzer/cache/http/
    ~/.ort/analyzer/cache/remote_artifacts/
    ~/.ort/scanner/cache/http/

over

    ~/.ort/cache/analyzer/http/
    ~/.ort/cache/analyzer/remote_artifacts/
    ~/.ort/cache/scanner/http/

What about you, @mnonnenmacher?

**Comment by mnonnenmacher** [comment]
I also prefer the proposed format, I'll update the PR.

File: scanner/src/main/kotlin/ArtifactoryCache.kt [hunk 1]
```diff
@@ -36,6 +36,9 @@ class ArtifactoryCache(
         private val url: String,
         private val apiToken: String
 ) : ScanResultsCache {
+    companion object {
+        private val HTTP_CACHE_PATH = "scanner/cache/http"
```
Comment (sschuberth) [inline scanner/src/main/kotlin/ArtifactoryCache.kt]: `@JvmStatic` also here?

Comment (mnonnenmacher) [reply scanner/src/main/kotlin/ArtifactoryCache.kt]: Done.

**Comment by sschuberth** [comment]
Oh, @mnonnenmacher, could you please adjust the cache entries in `.travis.yml` and `appveyor.yml` accordingly, too?

**Comment by mnonnenmacher** [comment]
@sschuberth Do we also want to archive the http caches? So far only the remote artifacts are archived.

**Comment by sschuberth** [comment]
In the language of Travis and AppVeyor, currently nothing is archived. Travis can only archive to external S3, and for AppVeyor I've disabled archiving in e9a186aa0d6ec18c64841ae5bac909f8b853fa4a. But I guess what you mean is the saving & restoring of cached directories. And yes, I don't think there's anything wrong with also caching the http caches. Maybe these directives support wildcards to easy specifying the different directories?

**Comment by mnonnenmacher** [comment]
Done, but I couldn't find anything about wildcards in the docs. I have added the paths to all caches separately for full control.

**Pull request merged**